.github/workflows/shellcheck-posix.yml

```diff
@@ -39,7 +39,7 @@ jobs:
       - name: check
         run: |
           echo "$GITHUB_WORKSPACE/shellcheck-v${SHELLCHECK_VERSION}" >> $GITHUB_PATH
-          readarray -t SCRIPT_FILES < <(grep -ErIl '^#![[:blank:]]*/bin/(ba|da|c|k)?sh' "$GITHUB_WORKSPACE/maven-wrapper-distribution/src/resources")
+          readarray -t SCRIPT_FILES < <(grep -ErIl -e '^#![[:blank:]]*/bin/(ba|da|c|k)?sh' -e '^#!/usr/bin/env[[:blank:]]+(ba|da|c|k)?sh' "$GITHUB_WORKSPACE/maven-wrapper-distribution/src/resources")
```

Comment:
should it also match `zsh`?

Reply:
Can do. But then, we should only write POSIX scripts so it runs everywhere. This would only be needed for shell specific scripts. I'd consider it optional, but will add it later.

Thanks for the finding.